### ISSUE TITLE
Java.Lang.IndexOutOfBoundsException: 'setSpan ( ... ) ends beyond length

### ISSUE THREAD

**Issue by Delphinidae84** [dotnet/maui#19689]
### Description

`Entry`-Elements cause a `Java.Lang.IndexOutOfBoundsException` when the binding contains more characters than specified by the `MaxLength`-Attribute.

### View

```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
             xmlns:local="clr-namespace:maui_issue_samples"
             x:Class="maui_issue_samples.MainPage">
    <ContentPage.BindingContext>
        <local:DemoViewModel />
    </ContentPage.BindingContext>

    <Entry Text="{Binding Text}" MaxLength="2"/>
</ContentPage>
```

### ViewModel
```

using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace maui_issue_samples
{
    class DemoViewModel : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler? PropertyChanged;

        public String Text
        {
            get => _text;
            set
            {
                if (value != _text)
                {
                    _text = value;
                    OnPropertyChanged();
                }
            }
        }
        private String _text = String.Empty;

        public DemoViewModel()
        {
            Text = "Bla";
        }

        public void OnPropertyChanged([CallerMemberName] string name = "") =>
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));

    }

}
```

![grafik](https://github.com/dotnet/maui/assets/38644741/dce6f91c-1d0a-44e7-838f-a9f540d0788e)

This issue was supposed to be fixed in #9418, it is easily reproducible that it was not though. See reproduction project below (use `entry-crash`-branch)

### Steps to Reproduce

1. Create a blank MAUI project
2. Add an Entry element like this one: `<Entry Text="{Binding Text}" MaxLength="2"/>`
3. Set the `Text` binding to a `String` with a length > 2 characters in the according ViewModel
4. Run on Android
5. App crashes 💥 

### Link to public reproduction project repository

https://github.com/Delphin-Technology/maui-issue-samples/tree/entry-crash

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

**Comment by jsuarezruiz** [comment]
The issue will be fixed by https://github.com/dotnet/maui/tree/fix-19689

**Comment by SchallerStefan** [comment]
When will this be merged into a release? I am in dire need of this but it was not in the last release on Feb. 14.

**Comment by shawncothran** [comment]
@jsuarezruiz @PureWeen is there a timeline for prioritizing this? Watching the streams for hope!

**Comment by david-maw** [comment]
I've been seeing the same symptom in Android Play Store testing of my App so I don't *think* it is only caused by code behind sending too long a value. Unfortunately, I cannot reproduce it so far, but I suspect it's caused by Play Store random input testing. The problem went away when I eliminated the MaxLength value from the half dozen or so entry fields that used it, so that's pretty good circumstantial evidence that it's a related issue.

**Comment by Sztub** [comment]
Is there any chance that fix will be included in next release? According to @jsuarezruiz  It has been fixed on Jan 16 2024...

**Comment by SchallerStefan** [comment]
Why was this completed on August 12 (even pushed into main), but not included in the recent release 8.0.82 SR8.2 from Aug. 29?
Or do we have to wait for the next release?

**Comment by dartasen** [comment]
@SchallerStefan It's planned to be released in SR9
